src/Microsoft.AspNetCore.Rewrite/Internal/IISUrlRewrite/IISUrlRewriteRule.cs

```diff
@@ -57,6 +61,32 @@ public virtual void ApplyRule(RewriteContext context)
                 }
             }
 
+            if (ServerVariables != null)
+            {
+                foreach (ServerVariable serverVariable in ServerVariables)
+                {
+                    var name = serverVariable.Name;
+                    var value = serverVariable.Evaluate(context, initMatchResults, condMatchRes);
+
+                    IHeaderDictionary headerDictionary;
+                    switch (serverVariable.Type)
+                    {
+                        case ServerVariableType.Request:
+                            headerDictionary = context.HttpContext.Request.Headers;
+                            context.Logger?.RequestHeaderAdded(name, value);
+                            break;
+                        case ServerVariableType.Response:
+                            headerDictionary = context.HttpContext.Response.Headers;
+                            context.Logger?.ResponseHeaderAdded(name, value);
+                            break;
+                        default:
+                            throw new ArgumentOutOfRangeException($"Unexpected server variable: `{serverVariable.Type}`");
+                    }
+
+                    headerDictionary.Append(name, value);
```

Comment:
There's an option in the IIS Url Rewrite module which is set by default to replace the existing request header value if it already exists. Otherwise it seems as if it leaves it alone.